src/main/scala/strawman/collection/Set.scala

```diff
@@ -42,9 +58,13 @@ trait SetLike[A, +C[X] <: Set[X]]
 trait SetMonoTransforms[A, +Repr]
   extends IterableMonoTransforms[A, Repr] {
 
-  def & (that: Set[A]): Repr
+  def & (that: Set[A]): Repr = this.filter(that)
+
+}
+
+trait SetPolyTransforms[A, +C[X]] extends IterablePolyTransforms[A, C] {
 
-  def ++ (that: Set[A]): Repr
+  def ++ (that: Set[A]): C[A]
```

Comment:
I don't see why this would be unsound. I don't want to open up the can of worms with intentional vs extensional sets again but no matter which one you pick (which determines variance) we still have universal equality and hashing, A default implementation with a typecast may not be correct in all cases (`SortedSet` could be a problem here) but `Set[A]` -> `Iterable[A]` -> `Iterable[B]` -> `Set[B]` is always well defined, so there must always be a way to implement this operation.